### PR TITLE
Brought structure up to date + Added analysers and cleaned up errors

### PR DESCRIPTION
**Pull request by kaderhashmi**

### CODE REVIEW

File: DFC.Composite.HealthMonitor.Data/DFC.Composite.HealthMonitor.Data.csproj [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.2</TargetFramework>
```
Comment (Muthuramana) [inline DFC.Composite.HealthMonitor.Data/DFC.Composite.HealthMonitor.Data.csproj]: Might as well upgrade to 3.1?

File: DFC.Composite.HealthMonitor.Services/HealthMonitoring/HealthMonitoringProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            var paths = await pathService.GetPaths().ConfigureAwait(false);
+
+            foreach (var path in paths)
+            {
+                var regions = await regionService.GetRegions(path.Path).ConfigureAwait(false);
+
+                foreach (var region in regions)
+                {
+                    if (!region.RequiresHealthCheck())
```
Comment (Muthuramana) [inline DFC.Composite.HealthMonitor.Services/HealthMonitoring/HealthMonitoringProcessor.cs]: Can we not use linq to filter out?

File: DFC.Composite.HealthMonitor.Data/DFC.Composite.HealthMonitor.Data.csproj (hunk 1 above)
Comment (kaderhashmi) [reply DFC.Composite.HealthMonitor.Data/DFC.Composite.HealthMonitor.Data.csproj]: Upgraded to net core 3.1

File: DFC.Composite.HealthMonitor.Services/HealthMonitoring/HealthMonitoringProcessor.cs (hunk 2 above)
Comment (kaderhashmi) [reply DFC.Composite.HealthMonitor.Services/HealthMonitoring/HealthMonitoringProcessor.cs]: Filtered out as suggested

**Pull request merged**